sdk/nodejs/runtime/closure/codePaths.ts

```diff
@@ -200,9 +200,18 @@ function allFoldersForPackages(
                     }
                 }
 
+                // package.json files can contain circularities.  For example es6-iterator depends
+                // on es5-ext, which depends on es6-iterator, which depends on es5-ext:
+                // https://github.com/medikoo/es6-iterator/blob/0eac672d3f4bb3ccc986bbd5b7ffc718a0822b74/package.json#L20
+                // https://github.com/medikoo/es5-ext/blob/792c9051e5ad9d7671dd4e3957eee075107e9e43/package.json#L29
+                //
+                // So keep track of the paths we've looked and don't recurse if we hit something again.
+                const seenPaths = new Set<string>();
+
                 const normalizedPackagePaths = new Set<string>();
```

Comment:
note: i considered reusing this.  however, there are codepaths we follow that don't necessary add a path to this.  while unlikely, you could maybe get into a circularity there.  so i use two different sets to prevent that concern.